common/workunit/workunit.cpp

```diff
@@ -6805,6 +6863,140 @@ const char *clusterTypeString(ClusterType clusterType, bool lcrSensitive)
     throwUnexpected();
 }
 
+bool extractFromWorkunitDAToken(const char * distributedAccessToken, StringBuffer * wuid, StringBuffer * user, StringBuffer * privKey)
```

Comment:
Using a bunch of strstr calls doesn't provide as much validation as I was hopping for.  At least in the security manager If the token isn't formatted right we may switch to treating it as a real password therefore we care that the format is exactly right.  I'd like to see a generic shared way of parsing out all of the attributes of a token and if the format isn't right failing gracefully.

Reply:
I create the token in the correct format, not sure what benefit would come of that. If someone malicious creates one in the wrong form, it will fail as it should